lib/streamlit/elements/widgets/file_uploader.py

```diff
@@ -37,11 +36,12 @@
     WidgetKwargs,
     register_widget,
 )
-from streamlit.runtime.uploaded_file_manager import UploadedFile, UploadedFileRec
+from streamlit.runtime.uploaded_file_manager import DeletedFile, UploadedFile
 from streamlit.type_util import Key, LabelVisibility, maybe_raise_label_warnings, to_key
 
-SomeUploadedFiles = Optional[Union[UploadedFile, List[UploadedFile]]]
-
+SomeUploadedFiles = Optional[
+    Union[UploadedFile, DeletedFile, List[Union[UploadedFile, DeletedFile]]]
+]
```

Comment:
nit: can add `None` to the union instead of having an `Optional[Union[...]]` to be consistent with camera input

Reply:
done